includes/Admin.php

```diff
@@ -35,19 +35,29 @@
  * Admin class.
  */
 class Admin {
+
+	/**
+	 * @var Plugin
+	 */
+	public $plugin;
+
+	public function __construct( $plugin ) {
```

Comment:
Only pass in the plugin object, when you need method, variables from it or any other sub classes. In this case we need something out of media, so we will pass the class in.